mcos/optimizer.py

```diff
@@ -1,17 +1,18 @@
 from __future__ import division
+
+from abc import ABC, abstractmethod
 from typing import Dict, List
 
 import numpy as np
-from abc import ABC, abstractmethod
-from pypfopt.efficient_frontier import EfficientFrontier
 import pandas as pd
-from sklearn.cluster import KMeans
-from sklearn.metrics import silhouette_samples
 import scipy.cluster.hierarchy as sch
+from numpy.linalg import inv, pinv
+from pypfopt.efficient_frontier import EfficientFrontier
 from scipy.optimize import minimize
-from matplotlib import pyplot as plt
```

Comment:
`matplotlib` was an unused import so I removed it as it was causing errors as it wasn't in `requirements.txt`